### PR TITLE
Make resource server paths configurable

### LINKED ISSUE #7: Security module configuration
Hard coded constants in the security module should be configurable.

### PR DESCRIPTION
**Pull request by sluehr**
Closes everest-engineering/lhotse#7

### CODE REVIEW

**Comment by sluehr** [comment]
This makes the most commonly modified parameters configurable. It doesn't address all the hard coded values and no longer feels urgent since we have to migrate to Spring's new [authorisation server](https://github.com/spring-projects-experimental/spring-authorization-server) at some point so this will all be rewritten :-P

File: src/main/java/engineering/everest/starterkit/security/config/ResourceServerConfig.java [hunk 1]
```diff
@@ -21,51 +22,27 @@
...
 
     private final JwtAccessTokenConverter jwtAccessTokenConverter;
 
     @Autowired
     public ResourceServerConfig(JwtAccessTokenConverter jwtAccessTokenConverter,
-                                ApplicationUserDetailsService userDetailsService) {
+                                ApplicationUserDetailsService userDetailsService,
+                                @Value("${application.security.endpoint.matchers.anonymous}") String[] anonymousUserAntPaths,
+                                @Value("${application.security.endpoint.matchers.authenticated}") String[] authenticatedUserAntPaths,
+                                @Value("${application.security.endpoint.matchers.admin}") String[] adminUserAntPaths) {
```
Comment (ywangd) [inline src/main/java/engineering/everest/starterkit/security/config/ResourceServerConfig.java:36]: Don't we need config properties for these values?

**Review by ywangd — COMMENTED** [review]
I left a question

File: src/main/java/engineering/everest/starterkit/security/config/ResourceServerConfig.java (hunk 1 above)
Comment (sluehr) [reply src/main/java/engineering/everest/starterkit/security/config/ResourceServerConfig.java:36]: Yes, config values would go into the main project.

Are you suggesting that we provide defaults here?

Comment (ywangd) [reply src/main/java/engineering/everest/starterkit/security/config/ResourceServerConfig.java:36]: Ah sorry I didn't realise it is in a separate project. You are right that it should in the main project, not here.

**Review by ywangd — APPROVED** [review]
LGTM

**Pull request merged**